### PR TITLE
Log Records Pagination Rendering

### PR DESCRIPTION
**Pull request by achandrabalan**
## Notion task link

[Log Records Pagination Rendering](https://www.notion.so/uwblueprintexecs/Log-Records-Pagination-Rendering-4511ae2d72fd4a3ab171e9edb662895f)

## Implementation description
* Deleted log records are immediately removed from modal, not after refresh
* Deleting log records appropriately adjusts pagination and user page number

## Steps to test
Deleting a record ..
1. Decrements the total record count by one
2. Initiates toast confirmation message
3. Immediately removes log from modal w/o refresh
4. If it is the last record on the page, the user is brought to the previous page. IE (User was on page 4, deletes the last record there and is now on page 3)
5. If it is not the last record on the page, then the user remains on the current page and all records below the deleted one are shifted up

## What should reviewers focus on?
* If there are no logs, should the user see 'Page 0 of 0' or 'Page 1 of 1' ?

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [ ] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [ ] I have documented this PR in the [Production Release Page](https://www.notion.so/uwblueprintexecs/fe28a97bfa5648d3bc3795b32b694acd?v=5565cfb35dcc45bb84f6528cb09517cd)
- [ ] I have updated other Docs as needed

### CODE REVIEW

**Comment by achandrabalan** [comment]
Updated PR to return to 'Page 0 of 0' base case when there are no records to display.

**Review by owen-sellner — APPROVED** [review]
LGTM!!!

**Review by phamkelly17 — APPROVED** [review]

**Review by connor-bechthold — CHANGES_REQUESTED** [review]
Looks great! One thing I'm realizing now going through this is I think it makes more sense to keep the page number on the current page when editing (instead of going back to 1)

You'll find what you're looking for in `EditLog.tsx` on lines 266 and 267 - you'll need to pass in the current page number as a prop like you did here as well

File: frontend/src/components/pages/HomePage/LogRecordsTable.tsx [hunk 1]
```diff
@@ -117,7 +119,15 @@ const LogRecordsTable = ({
       return;
     }
     handleDeleteToggle(itemId);
+    const newUserPageNum: number = (
```
Comment (connor-bechthold) [inline frontend/src/components/pages/HomePage/LogRecordsTable.tsx]: Nit: You can remove the `number` type here

File: frontend/src/components/pages/HomePage/LogRecordsTable.tsx [hunk 2]
```diff
@@ -117,7 +119,15 @@ const LogRecordsTable = ({
       return;
     }
     handleDeleteToggle(itemId);
```
Comment (connor-bechthold) [inline frontend/src/components/pages/HomePage/LogRecordsTable.tsx]: Nit: Delete toggle should be the last thing called in this sequence

**Review by connor-bechthold — APPROVED** [review]
LGTM 🧨

**Review by connor-bechthold — APPROVED** [review]
LGTM 🚒

**Pull request merged**